src/ddmd/hdrgen.d

```diff
@@ -1754,7 +1754,7 @@ public:
 
     override void visit(ClassDeclaration d)
     {
-        if (!d.isAnonymous())
```

Comment:
I count to 28 usages of the `isAnonymous` method. Why is only this one changed? Perhaps the implementation of the method should be updated to use the new enum.

Reply:
`isAnonymous` is a final method in the Dsymbol AST node. A ClassDeclaration inherits a Dsymbol so changing isAnonymous everywhere will break code. What we can do is change the check for class declarations or make `isAnonymous` not final and override it in the ClassDeclaration.

Reply:
of the 28 usages, only 1 is actually from a class declaration and I updated the code.

Reply:
Ah, ok, I actually didn't check how it was used, I only counted the occurrences. But I feel there's a risk of using the wrong way to identify if a class is an anonymous class when there are both the method `isAnonymous` and the instance variable `classKind`. Perhaps it's best to override it in `ClassDeclaration`, just to be on the safe side.